### PR TITLE
synth repair

### PR DESCRIPTION
**Pull request by grinkgo**
## About The Pull Request

Changes the behavior of a synthetic using cable coil and/or welders on themselves. They can now repair their own brute and burn damage after a short delay. Having someone else repair them is still faster, though.

## Why It's Good For The Game

It gives synthetic players a bare amount of self-sufficiency, which can reduce their frustration and dependency on other players, especially during lowpop rounds where there may not be any engineers. The player experience is better overall because instead of not being able to repair themselves at all, they have the ability to albeit at a slow rate. It's important to have that self-sufficiency without eliminating any need for other players at all.

## Changelog
:cl:
tweak: Synth players rejoice! You no longer have to rely on pesky humans to repair your own body (however, they still do it faster than you can).
soundadd: repairing synths with a welder now makes the welding noise.
/:cl:

### CODE REVIEW

File: code/game/objects/items/tools/maintenance_tools.dm [hunk 1]
```diff
@@ -202,20 +202,19 @@
...
 			S.heal_damage(15,0,0,1)
 			H.UpdateDamageIcon()
 			user.visible_message("<span class='warning'>\The [user] patches some dents on \the [H]'s [S.display_name] with \the [src].</span>", \
 								"<span class='warning'>You patch some dents on \the [H]'s [S.display_name] with \the [src].</span>")
 			remove_fuel(1,user)
+			playsound(user.loc, 'sound/items/Welder2.ogg', 25, 1)
 			return
 		else
-			to_chat(user, "<span class='warning'>Nothing to fix!</span>")
+			to_chat(user, "Nothing to fix!")
```
Comment (DominikPanic) [inline code/game/objects/items/tools/maintenance_tools.dm]: Don't remove useful formatting for no reason

File: code/game/objects/items/stacks/cable_coil.dm [hunk 2]
```diff
@@ -118,20 +118,20 @@
     
 /obj/item/stack/cable_coil/attack(mob/M as mob, mob/user as mob)
 	if(hasorgans(M))
+		var/mob/living/carbon/human/H = M
 		var/datum/limb/S = M:get_limb(user.zone_selected)
+
+		if(!S) return
```
Comment (DominikPanic) [inline code/game/objects/items/stacks/cable_coil.dm]: Single line control statement (press enter after if(!S))

File: code/game/objects/items/tools/maintenance_tools.dm (hunk 1 above)
Comment (grinkgo) [reply code/game/objects/items/tools/maintenance_tools.dm]: I just did that to match the cable coil message

Comment (DominikPanic) [reply code/game/objects/items/tools/maintenance_tools.dm]: Then you should add formatting to that. Making the experience worse is never the correct choice.

Comment (grinkgo) [reply code/game/objects/items/tools/maintenance_tools.dm]: gotcha

File: code/game/objects/items/stacks/cable_coil.dm (hunk 2 above)
Comment (grinkgo) [reply code/game/objects/items/stacks/cable_coil.dm]: :ok_hand:

File: code/game/objects/items/stacks/cable_coil.dm [hunk 3]
```diff
@@ -118,23 +118,24 @@
     
 /obj/item/stack/cable_coil/attack(mob/M as mob, mob/user as mob)
 	if(hasorgans(M))
```
Comment (spookydonut) [inline code/game/objects/items/stacks/cable_coil.dm]: this is conceptually haslimbs(), but for now it should stay like this given theres plans to give non-humans limbs at some point.

**Pull request merged**